core/src/com/unciv/models/translations/TranslationFileWriter.kt

```diff
@@ -17,99 +18,130 @@ import java.lang.reflect.Field
+                stringBuilder.appendln(lineToWrite)
             }
-            val translationKey = line.split(" = ")[0].replace("\\n","\n")
-            var translationValue = ""
-            if(translations.containsKey(translationKey)) translationValue = translations[translationKey]!!
-            else stringBuilder.appendln(" # Requires translation!")
-            val lineToWrite = translationKey.replace("\n","\\n") +
-                    " = "+ translationValue.replace("\n","\\n")
-            stringBuilder.appendln(lineToWrite)
-        }
-        Gdx.files.local("jsons/translations/$language.properties")
-                .writeString(stringBuilder.toString(),false, TranslationFileReader.charset)
-    }
 
+            countOfTranslatedLines[language] = translationsOfThisLanguage
 
-    fun writeNewTranslationFiles(translations: Translations) {
+            val fileWriter = if (modFolder != null) modFolder.child(languageFileLocation.format(language))
+                                else Gdx.files.local(languageFileLocation.format(language))
+            fileWriter.writeString(stringBuilder.toString(), false, TranslationFileReader.charset)
+        }
 
-        for (language in translations.getLanguages()) {
-            val languageHashmap = HashMap<String, String>()
+        // Calculate the percentages of translations
+        // It should be done after the loop of languages, since the countOfTranslatableLines is not known in the 1st iteration
+        for (key in countOfTranslatedLines.keys)
+            countOfTranslatedLines[key] = countOfTranslatedLines.getValue(key)*100/countOfTranslatableLines
 
-            for (translation in translations.values) {
-                if (translation.containsKey(language))
-                    languageHashmap[translation.entry] = translation[language]!!
-            }
-            writeByTemplate(language, languageHashmap)
-        }
-        writeLanguagePercentages(translations)
+        return countOfTranslatedLines
     }
 
-    private fun writeLanguagePercentages(translations: Translations){
-        val percentages = translations.calculatePercentageCompleteOfLanguages()
+    private fun writeLanguagePercentages(percentages: HashMap<String,Int>){
         val stringBuilder = StringBuilder()
         for(entry in percentages){
             stringBuilder.appendln(entry.key+" = "+entry.value)
         }
         Gdx.files.local(TranslationFileReader.percentagesFileLocation).writeString(stringBuilder.toString(),false)
     }
 
-    private fun generateTutorialsStrings(): Collection<String> {
 
-        if (generatedStrings.containsKey("Tutorials"))
-            return generatedStrings.getValue("Tutorials")
 
-        generatedStrings["Tutorials"] = mutableSetOf()
-        val tutorialsStrings = generatedStrings["Tutorials"]
+    private fun generateTutorialsStrings(): MutableSet<String> {
+
+        val tutorialsStrings = mutableSetOf<String>()
```

Comment:
If you want them to be in the samee order you put them in, this should be a LinkedHashSet

Reply:
That's true, and it is, because `mutableSetOf` is declared as:
> public inline fun <T> mutableSetOf(): MutableSet<T> = LinkedHashSet()